webauthn/Sources/App/Controllers/WebAuthnController.swift

```diff
@@ -20,167 +20,126 @@ import HummingbirdCore
 import WebAuthn
 
 struct HBWebAuthnController {
+    let webauthn: WebAuthnManager
+
     func add(_ group: HBRouterGroup) {
         group
-            .post("signup", options: .editResponse, use: SignupHandler.self)
+            .post("signup", options: .editResponse, use: self.signin)
             .get("login", options: .editResponse, use: self.beginAuthentication)
+        group
+            .add(middleware: WebAuthnSessionStateAuthenticator())
+            .post("beginregister", use: self.beginRegistration)
+            .post("finishregister", use: self.finishRegistration)
+            .post("login", options: .editResponse, use: self.finishAuthentication)
         group
             .add(middleware: WebAuthnSessionAuthenticator())
-            .post("beginregister", use: BeginRegistrationHandler.self)
-            .post("finishregister", use: FinishRegistrationHandler.self)
-            .post("login", options: .editResponse, use: FinishAuthenticationHandler.self)
-            .get("test", use: self.authenticationDetails)
+            .get("test", use: self.getUser)
     }
 
-    struct SignupHandler: HBAsyncRouteHandler {
-        struct Input: Decodable {
-            let name: String
-        }
-
-        typealias Output = HBResponse
-
-        let input: Input
-
-        init(from request: HBRequest) throws {
-            self.input = try request.decode(as: Input.self)
-        }
+    struct SignInInput: Decodable {
+        let name: String
+    }
 
-        func handle(request: HBRequest) async throws -> Output {
-            guard try await User.query(on: request.db)
-                .filter(\.$username == self.input.name)
-                .first() == nil
-            else {
-                throw HBHTTPError(.conflict, message: "Username already taken.")
-            }
-            let user = User(username: self.input.name)
-            try await user.save(on: request.db)
-            let session = WebAuthnSessionAuthenticator.Session.signedUp(userId: user.id!)
-            try await request.session.save(
-                session: session,
-                expiresIn: .minutes(10)
-            )
-            return .redirect(to: "/api/beginregister", type: .temporary)
+    func signin(request: HBRequest) async throws -> HBResponse {
+        let input = try request.decode(as: SignInInput.self)
+        guard try await User.query(on: request.db)
+            .filter(\.$username == input.name)
+            .first() == nil
+        else {
+            throw HBHTTPError(.conflict, message: "Username already taken.")
         }
+        let user = User(username: input.name)
+        try await user.save(on: request.db)
+        let session = WebAuthnSessionStateAuthenticator.Session.signedUp(userId: user.id!)
```

Comment:
I know you just saved the user, but especially in an example it's not good to force unwrap variables. Users _may_ define their IDs as user generated, and run into a confusing crash that way. Use `.requireId()` instead, which I believe is a FluentKit API